Qemu 2.10 requires `-U` flag to read volume metadata

GutoVeronezi opened the pull request:
### Description
`Qemu 2.10 added the requirement of a --force-share flag to 'qemu-img info' command when reading information about a disk that is in use by a guest`. After this Qemu release, to obtain the attached disk metadata, we need to use a flag to force information reading.   

This PR intends to add the required flag to the `qemu-img info` command if the qemu version is 2.10+. We can get the Qemu version in the LibvirtConnection.

Fixes: #4024

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### Feature/Enhancement Scale or Bug Severity


#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

### How Has This Been Tested?
It has been tested locally.

rhtyd commented:
@GutoVeronezi what error/issue do you see and on which KVM distro?

rafaelweingartner commented:
@rhtyd I checked this one with him. The problem happens on Qemu 2.10+. The release notes from Qemu have the following message.
```
Qemu 2.10 added the requirement of a --force-share flag to qemu-img
info when reading information about a disk that is in use by a
guest. We do this a lot in Nova for operations like gathering
information before live migration.

```

The problem is happening when ACS tries to get some info from a volume using qemu-img. Then, one would get the following error.
```
qemu-img: Could not open '/mnt/.../../volumeID': Failed to get shared "write" lock
Is another process using the image?

```

rhtyd commented:
Okay but which KVM/Linux distro is it reproducible on?

GutoVeronezi commented:
It has been tested over Ubuntu 18.04 with KVM 2.11.1

weizhouapache reviewed (commented):
code lgtm
@DaanHoogland @rhtyd we need this in 4.15.0.0 as well.

rhtyd commented:
Looks like RC4 has been cut, given Ubuntu 18.04 has been around for quite some time should this be considered a blocker?

blueorangutan commented:
Packaging result: ✔centos7 ✔centos8 ✔debian. JID-2526

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rhtyd commented:
@GutoVeronezi @weizhouapache probably too late now - RC4 is cut; in order to get the PR into 4.15, the base branch needs to be changed

GutoVeronezi commented:
Thanks for the review guys. To us, it is fine if it only goes to master.

weizhouapache commented:
> Thanks for the review guys. To us, it is fine if it only goes to master.

ok, let's merge this after 4.15.0.0 release

RodrigoDLopez reviewed (approved):
Code LGTM...
I think we need Unit tests to cover these changes... But to do this you'll need to do some changes in many methods. 
Maybe you can do this in another PR?

Another point ... maybe you can better explain how this branch can be tested. 
> It has been tested locally.

Your explanation does not tell us anything related to the tests.

GutoVeronezi commented:
@RodrigoDLopez to test this enhancement I had used the `create snapshot` API.
With my test instance running, I tried to take a snapshot from the volume. Whitout the `-u` flag I was getting `qemu-img: Could not open '/mnt/...': Failed to get shared "write" lockIs another process using the image?` error.

GabrielBrascher reviewed (approved):
Code LGTM.

However, considering that this impacts versions that have at least ubuntu 18.04 (if I am not mistaken Ubuntu 18.04 gets by default qemu-kvm 2.11). It would be nice to have this on prior LTSs.

Any comment on what branches/releases should we target this one? @rhtyd @weizhouapache @rafaelweingartner and others

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos7 ✖centos8 ✖debian. JID-2615

DaanHoogland commented:
@blueorangutan package

blueorangutan commented:
@DaanHoogland a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos7 ✔centos8 ✔debian. JID-2617

DaanHoogland commented:
@blueorangutan test

blueorangutan commented:
@DaanHoogland a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-3448)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 42227 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t3448-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Smoke tests completed. 84 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_deploy_kubernetes_cluster | `Failure` | 3624.07 | test_kubernetes_clusters.py
test_04_deploy_and_scale_kubernetes_cluster | `Failure` | 22.51 | test_kubernetes_clusters.py
test_07_deploy_kubernetes_ha_cluster | `Failure` | 26.61 | test_kubernetes_clusters.py
test_08_deploy_and_upgrade_kubernetes_ha_cluster | `Failure` | 0.04 | test_kubernetes_clusters.py
test_09_delete_kubernetes_ha_cluster | `Failure` | 0.04 | test_kubernetes_clusters.py
ContextSuite context=TestKubernetesCluster>:teardown | `Error` | 78.68 | test_kubernetes_clusters.py
test_04_rvpc_privategw_static_routes | `Failure` | 868.50 | test_privategw_acl.py

weizhouapache commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java:
```diff
@@ -234,14 +235,19 @@ public void create(final QemuImgFile file, final Map<String, String> options) th
 
         script.add("convert");
+        Long version  = LibvirtConnection.getConnection().getVersion();
+        if (version > 2009000) {
```
@GutoVeronezi 
I checked qemu codebase, this is introduced in 2.10.0

so I suggest to use 
```
if (version >= 2010000) {
```

qemu 2.9.1 does not support it.

weizhouapache commented on plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java:
```diff
@@ -343,10 +349,15 @@ public void commit(final QemuImgFile file) throws QemuImgException {
         final Script s = new Script(_qemuImgPath);
         s.add("info");
+        Long version  = LibvirtConnection.getConnection().getVersion();
+        if (version > 2009000) {
```
@GutoVeronezi same here.

weizhouapache commented:
It seem this does not impact NFS-based environments, as they use script files in scripts/storage/qcow2/ to manage volumes/template/snapshots.

I do not know if there are critical issues with RBD-based environments. maybe @wido @GabrielBrascher know.

wido commented:
> It seem this does not impact NFS-based environments, as they use script files in scripts/storage/qcow2/ to manage volumes/template/snapshots.
> 
> I do not know if there are critical issues with RBD-based environments. maybe @wido @GabrielBrascher know.

NFS is also impacted if you have enabled file locking (which most NFS envs have enabled).

RBD isn't impacted in any case as it doesn't use QCOW2.

DaanHoogland commented:
> Trillian test result (tid-3448)
> Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
> Total time taken: 42227 seconds
> Marvin logs: [blueorangutan/acs-prs@`trillian` pr4567-t3448-kvm-centos7.zip (download)](https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t3448-kvm-centos7.zip)
> Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
> Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
> Smoke tests completed. 84 look OK, 2 have error(s)
> Only failed tests results shown below:
> Test 	Result 	Time (s) 	Test File
> test_01_deploy_kubernetes_cluster 	`Failure` 	3624.07 	test_kubernetes_clusters.py
> test_04_deploy_and_scale_kubernetes_cluster 	`Failure` 	22.51 	test_kubernetes_clusters.py
> test_07_deploy_kubernetes_ha_cluster 	`Failure` 	26.61 	test_kubernetes_clusters.py
> test_08_deploy_and_upgrade_kubernetes_ha_cluster 	`Failure` 	0.04 	test_kubernetes_clusters.py
> test_09_delete_kubernetes_ha_cluster 	`Failure` 	0.04 	test_kubernetes_clusters.py
> ContextSuite context=TestKubernetesCluster>:teardown 	`Error` 	78.68 	test_kubernetes_clusters.py
> test_04_rvpc_privategw_static_routes 	`Failure` 	868.50 	test_privategw_acl.py

@GutoVeronezi , can you check the private gateway error (you can ignore the kubernetes ones)?

weizhouapache commented:
> > It seem this does not impact NFS-based environments, as they use script files in scripts/storage/qcow2/ to manage volumes/template/snapshots.
> > I do not know if there are critical issues with RBD-based environments. maybe @wido @GabrielBrascher know.
> 
> NFS is also impacted if you have enabled file locking (which most NFS envs have enabled).
> 
 yes, the scripts in scripts/storage/qcow2/ has been changed to support -U in #4068
I have tested some actions on vms/volumes/templates on ubuntu 20.04, look ok without this pr.

> RBD isn't impacted in any case as it doesn't use QCOW2.
@wido as far as I know, this impacts RAW format as well.
`qemu.convert` is used for RBD in plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java
`q.convert` in plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

maybe there is no issue as in most cases vm is stopped.

weizhouapache commented:
> > Trillian test result (tid-3448)
> > Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
> > Total time taken: 42227 seconds
> > Marvin logs: [blueorangutan/acs-prs@`trillian` pr4567-t3448-kvm-centos7.zip (download)](https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t3448-kvm-centos7.zip)
> > Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
> > Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
> > Smoke tests completed. 84 look OK, 2 have error(s)
> > Only failed tests results shown below:
> > Test 	Result 	Time (s) 	Test File
> > test_01_deploy_kubernetes_cluster 	`Failure` 	3624.07 	test_kubernetes_clusters.py
> > test_04_deploy_and_scale_kubernetes_cluster 	`Failure` 	22.51 	test_kubernetes_clusters.py
> > test_07_deploy_kubernetes_ha_cluster 	`Failure` 	26.61 	test_kubernetes_clusters.py
> > test_08_deploy_and_upgrade_kubernetes_ha_cluster 	`Failure` 	0.04 	test_kubernetes_clusters.py
> > test_09_delete_kubernetes_ha_cluster 	`Failure` 	0.04 	test_kubernetes_clusters.py
> > ContextSuite context=TestKubernetesCluster>:teardown 	`Error` 	78.68 	test_kubernetes_clusters.py
> > test_04_rvpc_privategw_static_routes 	`Failure` 	868.50 	test_privategw_acl.py
> 
> @GutoVeronezi , can you check the private gateway error (you can ignore the kubernetes ones)?

@DaanHoogland it should have been fixed.

shwstppr commented:
@blueorangutan package

blueorangutan commented:
@shwstppr a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos7 ✔centos8 ✔debian. JID-2723

weizhouapache reviewed (approved):
code lgtm

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos7 ✖centos8 ✔debian. JID-2750

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos7 ✖centos8 ✖debian. JID-2765

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos7 ✖centos8 ✖debian. JID-2876

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos7 ✖centos8 ✖debian. JID-2884

wido commented:
Can you also fix this in *resizevolume.sh* ?

https://github.com/apache/cloudstack/blob/master/scripts/storage/qcow2/resizevolume.sh#L181

GutoVeronezi commented:
> Can you also fix this in _resizevolume.sh_ ?
> 
> https://github.com/apache/cloudstack/blob/master/scripts/storage/qcow2/resizevolume.sh#L181

@wido done.

GabrielBrascher reviewed (approved):
Code LGTM.

Manually tested the update on `resizevolume.sh`; it retrieves the correct qemu version and executes `qemu-img` command appending `-U` as expected.

GabrielBrascher commented:
@blueorangutan package

blueorangutan commented:
@GabrielBrascher a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress. [S]

blueorangutan commented:
Packaging result: :heavy_multiplication_x: centos7 :heavy_multiplication_x: centos8 :heavy_multiplication_x: debian. SL-JID 82

GabrielBrascher reviewed (commented):
All tests are failing due to:

```
[ERROR] /home/travis/build/apache/cloudstack/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:[4304,48] error: unreported exception LibvirtException; must be caught or declared to be thrown

[ERROR] /home/travis/build/apache/cloudstack/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java:[4310,63] error: unreported exception LibvirtException; must be caught or declared to be thrown
```

It is necessary to handle LibvirtException at `LibvirtComputingResource.java:4310`

GutoVeronezi commented:
@GabrielBrascher I rebased the PR and fixed it.

I will ask you to take a look at the commit db1e0f49dc06d16218ce61d9ef6a025075710d8e, which introduced the error you reported. I am wondering, how this commit was merged if Travis explicitly says that `The build canceled. This is a change from the previous build, which errored.`?

weizhouapache commented:
> @GabrielBrascher I rebased the PR and fixed it.
> 
> I will ask you to take a look at the commit [db1e0f4](https://github.com/apache/cloudstack/commit/db1e0f49dc06d16218ce61d9ef6a025075710d8e), which introduced the error you reported. I am wondering, how this commit was merged if Travis explicitly says that `The build canceled. This is a change from the previous build, which errored.`?

@GutoVeronezi the issue is caused by https://github.com/apache/cloudstack/pull/4567/files#diff-8c9bad5e83a17ce03945d04eea177a39d57f52b92c32cdc86888cedbc49d9a84R355
```
        Long version  = LibvirtConnection.getConnection().getVersion();
```
good you have fixed it.

GabrielBrascher commented:
@GutoVeronezi thanks for rebasing and fixing the compilation issue.

Answering your question:

You had correctly addressed most of the callers by adding `LibvirtException e` at their respective `catch` blocks. However, it was missing LibvirtException at the cath block in `LibvirtComputingResource.java`. Therefore, the error that I reported was a compilation error caused due to the addition of `LibvirtException` at the method's _throws_ that was not handled at LibvirtComputingResource.java (lines 4304 and 4310).

Commit that added the mentioned `LibvirtExeption`:
https://github.com/apache/cloudstack/blob/1c1d3566827ba75e5cf17a1c6fe0789a959a501d/plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/utils/qemu/QemuImg.java#L352

I don't understand how commit https://github.com/apache/cloudstack/commit/db1e0f49dc06d16218ce61d9ef6a025075710d8e would introduce that error. At that time method `QemuImg.info` was not throwing `LibvirtException`.

Additionally, checking at PR #4656 all looks good: Jenkins, Travis, and Trilian tests passed (https://github.com/apache/cloudstack/pull/4656/commits).

![image](https://user-images.githubusercontent.com/5025148/110800031-8da4fe00-825a-11eb-8c65-23f4273c5045.png)

GabrielBrascher commented:
@blueorangutan package

blueorangutan commented:
@GabrielBrascher a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress. [S]

blueorangutan commented:
Packaging result: :heavy_multiplication_x: centos7 :heavy_multiplication_x: centos8 :heavy_multiplication_x: debian. SL-JID 93

weizhouapache commented:
@GutoVeronezi  tested ok.
There are ignored NPE when migrate vm with datadisk. It is caused by commit db1e0f4 
Fixed by #4800

GabrielBrascher commented:
@GutoVeronezi Travis now points to `ScaleIOStorageAdaptor.java` (lines 271 & 365). Can you please re-check it?

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  09:17 min (Wall Clock)
[INFO] Finished at: 2021-03-11T15:59:17Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project cloud-plugin-hypervisor-kvm: Compilation failure: Compilation failure: 
[ERROR] /home/travis/build/apache/cloudstack/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/ScaleIOStorageAdaptor.java:[271,24] error: unreported exception LibvirtException; must be caught or declared to be thrown
[ERROR] /home/travis/build/apache/cloudstack/plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/ScaleIOStorageAdaptor.java:[365,24] error: unreported exception LibvirtException; must be caught or declared to be thrown
```

GutoVeronezi commented:
@GabrielBrascher done.

I looked for more places that needed to handle it, but found none.

GabrielBrascher commented:
Thanks, @GutoVeronezi!

GabrielBrascher reviewed (approved):
Tests indicate that all `LibvirtException` have been handled. LGTM.

blueorangutan commented:
Packaging result: :heavy_check_mark: centos7 :heavy_check_mark: centos8 :heavy_check_mark: debian. SL-JID 123

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress. [S]

blueorangutan commented:
Packaging result: :heavy_check_mark: centos7 :heavy_check_mark: centos8 :heavy_check_mark: debian. SL-JID 186

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress. [S]

blueorangutan commented:
Packaging result: :heavy_multiplication_x: centos7 :heavy_check_mark: centos8 :heavy_check_mark: debian. SL-JID 187

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests [S]

blueorangutan commented:
<b>[S] Trillian Build Failed (tid-184)<b/>

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress. [S]

blueorangutan commented:
Packaging result: :heavy_check_mark: centos7 :heavy_check_mark: centos8 :heavy_check_mark: debian. SL-JID 191

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests [S]

blueorangutan commented:
<b>[S] Trillian Build Failed (tid-185)<b/>

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests [S]

blueorangutan commented:
<b>[S] Trillian Build Failed (tid-186)<b/>

blueorangutan commented:
<b>[S] Trillian test result (tid-177)</b>
Environment: vmware-67u3 (x2), Advanced Networking with Mgmt server u18
Total time taken: 103108 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t177-vmware-67u3.zip
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_extra_config_data.py
Intermittent failure detected: /marvin/tests/smoke/test_diagnostics.py
Intermittent failure detected: /marvin/tests/smoke/test_human_readable_logs.py
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_supported_versions.py
Intermittent failure detected: /marvin/tests/smoke/test_loadbalance.py
Intermittent failure detected: /marvin/tests/smoke/test_nested_virtualization.py
Intermittent failure detected: /marvin/tests/smoke/test_network.py
Intermittent failure detected: /marvin/tests/smoke/test_nic.py
Intermittent failure detected: /marvin/tests/smoke/test_password_server.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dns.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dnsservice.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_iptables_default_policy.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermittent failure detected: /marvin/tests/smoke/test_routers.py
Intermittent failure detected: /marvin/tests/smoke/test_scale_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_router_nics.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Smoke tests completed. 58 look OK, 28 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_create_lb_rule_src_nat | `Failure` | 308.41 | test_loadbalance.py
test_02_create_lb_rule_non_nat | `Failure` | 34.79 | test_loadbalance.py
test_assign_and_removal_lb | `Failure` | 34.78 | test_loadbalance.py
test_01_port_fwd_on_src_nat | `Failure` | 606.03 | test_network.py
test_02_port_fwd_on_non_src_nat | `Failure` | 608.00 | test_network.py
test_reboot_router | `Failure` | 403.38 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | `Error` | 609.85 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | `Error` | 612.16 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | `Error` | 615.21 | test_network.py
test_01_deployVMInSharedNetwork | `Error` | 693.24 | test_network.py
test_02_verifyRouterIpAfterNetworkRestart | `Error` | 648.93 | test_network.py
test_03_destroySharedNetwork | `Failure` | 1.10 | test_network.py
ContextSuite context=TestSharedNetwork>:teardown | `Error` | 2.22 | test_network.py
test_05_deploy_vm_with_extraconfig_vmware | `Error` | 667.00 | test_deploy_vm_extra_config_data.py
test_01_nic | `Error` | 725.79 | test_nic.py
test_10_traceroute_in_vr | `Failure` | 60.59 | test_diagnostics.py
test_13_retrieve_vr_default_files | `Error` | 1.10 | test_diagnostics.py
test_14_retrieve_vr_one_file | `Error` | 1.12 | test_diagnostics.py
test_15_retrieve_ssvm_default_files | `Error` | 1.12 | test_diagnostics.py
test_16_retrieve_ssvm_single_file | `Error` | 1.16 | test_diagnostics.py
test_17_retrieve_cpvm_default_files | `Error` | 1.13 | test_diagnostics.py
test_18_retrieve_cpvm_single_file | `Error` | 1.13 | test_diagnostics.py
test_isolate_network_password_server | `Failure` | 157.46 | test_password_server.py
test_02_vpc_privategw_static_routes | `Failure` | 821.25 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 819.18 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Failure` | 947.24 | test_privategw_acl.py
test_01_disableHumanReadableLogs | `Error` | 604.46 | test_human_readable_logs.py
test_02_enableHumanReadableLogs | `Error` | 604.53 | test_human_readable_logs.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | `Failure` | 320.70 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | `Failure` | 390.48 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 195.18 | test_internal_lb.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 230.46 | test_internal_lb.py
test_01_deploy_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_02_invalid_upgrade_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_03_deploy_and_upgrade_kubernetes_cluster | `Failure` | 0.01 | test_kubernetes_clusters.py
test_04_deploy_and_scale_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_05_delete_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_06_deploy_invalid_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_07_deploy_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_08_deploy_and_upgrade_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_09_delete_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_01_add_delete_kubernetes_supported_version | `Error` | 0.01 | test_kubernetes_supported_versions.py
test_nested_virtualization_vmware | `Error` | 787.12 | test_nested_virtualization.py
test_router_dhcphosts | `Failure` | 157.69 | test_router_dhcphosts.py
ContextSuite context=TestRouterDHCPHosts>:teardown | `Error` | 168.12 | test_router_dhcphosts.py
test_router_dhcp_opts | `Error` | 612.20 | test_router_dhcphosts.py
test_router_dns_guestipquery | `Failure` | 455.36 | test_router_dns.py
test_router_dns_guestipquery | `Failure` | 455.41 | test_router_dnsservice.py
test_02_routervm_iptables_policies | `Error` | 686.82 | test_routers_iptables_default_policy.py
test_01_single_VPC_iptables_policies | `Error` | 772.37 | test_routers_iptables_default_policy.py
test_01_isolate_network_FW_PF_default_routes_egress_true | `Failure` | 234.79 | test_routers_network_ops.py
test_02_isolate_network_FW_PF_default_routes_egress_false | `Failure` | 237.26 | test_routers_network_ops.py
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | `Failure` | 297.39 | test_routers_network_ops.py
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | `Failure` | 289.86 | test_routers_network_ops.py
test_03_RVR_Network_check_router_state | `Error` | 734.54 | test_routers_network_ops.py
test_01_router_internal_basic | `Error` | 604.42 | test_routers.py
test_02_router_internal_adv | `Error` | 604.68 | test_routers.py
test_04_restart_network_wo_cleanup | `Error` | 606.58 | test_routers.py
test_01_scale_vm | `Error` | 604.50 | test_scale_vm.py
test_04_change_offering_small | `Failure` | 719.93 | test_service_offerings.py
test_01_snapshot_root_disk | `Error` | 650.75 | test_snapshots.py
test_03_ssvm_internals | `Error` | 604.61 | test_ssvm.py
test_04_cpvm_internals | `Error` | 604.72 | test_ssvm.py
test_05_stop_ssvm | `Error` | 662.47 | test_ssvm.py
test_06_stop_cpvm | `Error` | 695.58 | test_ssvm.py
test_07_reboot_ssvm | `Error` | 709.11 | test_ssvm.py
test_08_reboot_cpvm | `Error` | 612.08 | test_ssvm.py
test_09_reboot_ssvm_forced | `Error` | 758.78 | test_ssvm.py
test_10_reboot_cpvm_forced | `Error` | 678.37 | test_ssvm.py
test_11_destroy_ssvm | `Error` | 659.83 | test_ssvm.py
test_12_destroy_cpvm | `Error` | 665.92 | test_ssvm.py
ContextSuite context=TestVAppsVM>:setup | `Error` | 44.77 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | `Failure` | 609.15 | test_vm_life_cycle.py
test_change_service_offering_for_vm_with_snapshots | `Failure` | 692.91 | test_vm_snapshots.py
test_01_create_vm_snapshots | `Failure` | 604.51 | test_vm_snapshots.py
test_02_revert_vm_snapshots | `Failure` | 601.48 | test_vm_snapshots.py
test_03_delete_vm_snapshots | `Failure` | 0.02 | test_vm_snapshots.py
test_01_create_volume | `Failure` | 611.39 | test_volumes.py
test_02_attach_volume | `Failure` | 607.66 | test_volumes.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 915.88 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 909.73 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 0.48 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Error` | 752.99 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 896.78 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 945.23 | test_vpc_redundant.py
test_01_VPC_nics_after_destroy | `Failure` | 767.00 | test_vpc_router_nics.py
test_02_VPC_default_routes | `Failure` | 760.40 | test_vpc_router_nics.py
test_01_redundant_vpc_site2site_vpn | `Failure` | 456.13 | test_vpc_vpn.py
test_01_vpc_site2site_vpn_multiple_options | `Error` | 305.23 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Error` | 331.74 | test_vpc_vpn.py

blueorangutan commented:
<b>[S] Trillian test result (tid-181)</b>
Environment: xenserver-72 (x2), Advanced Networking with Mgmt server 7
Total time taken: 68625 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t181-xenserver-72.zip
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vgpu_enabled_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_global_settings.py
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_loadbalance.py
Intermittent failure detected: /marvin/tests/smoke/test_network.py
Intermittent failure detected: /marvin/tests/smoke/test_outofbandmanagement.py
Intermittent failure detected: /marvin/tests/smoke/test_password_server.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermittent failure detected: /marvin/tests/smoke/test_reset_vm_on_reboot.py
Intermittent failure detected: /marvin/tests/smoke/test_resource_accounting.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dns.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dnsservice.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_iptables_default_policy.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermittent failure detected: /marvin/tests/smoke/test_routers.py
Intermittent failure detected: /marvin/tests/smoke/test_scale_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_secondary_storage.py
Intermittent failure detected: /marvin/tests/smoke/test_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_templates.py
Intermittent failure detected: /marvin/tests/smoke/test_usage.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_router_nics.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Smoke tests completed. 48 look OK, 29 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
ContextSuite context=TestLoadBalance>:setup | `Error` | 0.00 | test_loadbalance.py
test_01_port_fwd_on_src_nat | `Failure` | 607.85 | test_network.py
test_02_port_fwd_on_non_src_nat | `Failure` | 608.98 | test_network.py
test_reboot_router | `Failure` | 955.80 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | `Error` | 609.23 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | `Error` | 610.18 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | `Error` | 613.24 | test_network.py
ContextSuite context=TestDeployvGPUenabledVM>:setup | `Error` | 0.00 | test_deploy_vgpu_enabled_vm.py
test_isolate_network_password_server | `Failure` | 161.29 | test_password_server.py
test_02_vpc_privategw_static_routes | `Failure` | 184.51 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 182.73 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Failure` | 275.41 | test_privategw_acl.py
test_UpdateConfigParamWithScope | `Error` | 0.13 | test_global_settings.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | `Failure` | 353.20 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | `Failure` | 373.27 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 193.16 | test_internal_lb.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 225.16 | test_internal_lb.py
ContextSuite context=TestResetVmOnReboot>:setup | `Error` | 0.00 | test_reset_vm_on_reboot.py
ContextSuite context=TestRAMCPUResourceAccounting>:setup | `Error` | 0.00 | test_resource_accounting.py
ContextSuite context=TestRouterDHCPHosts>:setup | `Error` | 0.00 | test_router_dhcphosts.py
ContextSuite context=TestRouterDHCPOpts>:setup | `Error` | 0.00 | test_router_dhcphosts.py
ContextSuite context=TestRouterDns>:setup | `Error` | 0.00 | test_router_dns.py
ContextSuite context=TestRouterDnsService>:setup | `Error` | 0.00 | test_router_dnsservice.py
ContextSuite context=TestRouterIpTablesPolicies>:setup | `Error` | 0.00 | test_routers_iptables_default_policy.py
ContextSuite context=TestVPCIpTablesPolicies>:setup | `Error` | 0.00 | test_routers_iptables_default_policy.py
ContextSuite context=TestIsolatedNetworks>:setup | `Error` | 0.00 | test_routers_network_ops.py
ContextSuite context=TestRedundantIsolateNetworks>:setup | `Error` | 0.00 | test_routers_network_ops.py
ContextSuite context=TestRouterServices>:setup | `Error` | 0.00 | test_routers.py
ContextSuite context=TestScaleVm>:setup | `Error` | 0.00 | test_scale_vm.py
test_01_sys_vm_start | `Failure` | 0.08 | test_secondary_storage.py
ContextSuite context=TestServiceOfferings>:setup | `Error` | 1511.26 | test_service_offerings.py
ContextSuite context=TestSnapshotRootDisk>:setup | `Error` | 0.00 | test_snapshots.py
test_01_list_sec_storage_vm | `Failure` | 0.03 | test_ssvm.py
test_02_list_cpvm_vm | `Failure` | 0.02 | test_ssvm.py
test_03_ssvm_internals | `Failure` | 0.02 | test_ssvm.py
test_04_cpvm_internals | `Failure` | 0.02 | test_ssvm.py
test_05_stop_ssvm | `Failure` | 0.03 | test_ssvm.py
test_06_stop_cpvm | `Failure` | 0.03 | test_ssvm.py
test_07_reboot_ssvm | `Failure` | 0.03 | test_ssvm.py
test_08_reboot_cpvm | `Failure` | 0.03 | test_ssvm.py
test_09_destroy_ssvm | `Failure` | 0.02 | test_ssvm.py
test_10_destroy_cpvm | `Failure` | 0.02 | test_ssvm.py
test_02_create_template_with_checksum_sha1 | `Error` | 65.35 | test_templates.py
test_03_create_template_with_checksum_sha256 | `Error` | 65.37 | test_templates.py
test_04_create_template_with_checksum_md5 | `Error` | 65.37 | test_templates.py
test_05_create_template_with_no_checksum | `Error` | 65.36 | test_templates.py
ContextSuite context=TestTemplates>:setup | `Error` | 1.97 | test_templates.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestLBRuleUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestNatRuleUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestPublicIPUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestSnapshotUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVmUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVolumeUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestVpnUsage>:setup | `Error` | 0.00 | test_usage.py
ContextSuite context=TestDeployVM>:setup | `Error` | 0.00 | test_vm_life_cycle.py
ContextSuite context=TestVMLifeCycle>:setup | `Error` | 1.67 | test_vm_life_cycle.py
test_change_service_offering_for_vm_with_snapshots | `Error` | 1.15 | test_vm_snapshots.py
ContextSuite context=TestVmSnapshot>:setup | `Error` | 8.89 | test_vm_snapshots.py
ContextSuite context=TestCreateVolume>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestVolumes>:setup | `Error` | 0.00 | test_volumes.py
ContextSuite context=TestVPCRedundancy>:setup | `Error` | 0.00 | test_vpc_redundant.py
ContextSuite context=TestVPCNics>:setup | `Error` | 0.00 | test_vpc_router_nics.py
ContextSuite context=TestRVPCSite2SiteVpn>:setup | `Error` | 0.00 | test_vpc_vpn.py
ContextSuite context=TestVPCSite2SiteVPNMultipleOptions>:setup | `Error` | 0.00 | test_vpc_vpn.py
ContextSuite context=TestVpcRemoteAccessVpn>:setup | `Error` | 0.00 | test_vpc_vpn.py
ContextSuite context=TestVpcSite2SiteVpn>:setup | `Error` | 0.00 | test_vpc_vpn.py
test_02_cancel_host_maintenace_with_migration_jobs | `Failure` | 186.55 | test_host_maintenance.py

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests [S]

blueorangutan commented:
<b>[S] Trillian test result (tid-188)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 104934 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t188-kvm-centos7.zip
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups_projects.py
Intermittent failure detected: /marvin/tests/smoke/test_affinity_groups.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_virtio_scsi_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_extra_config_data.py
Intermittent failure detected: /marvin/tests/smoke/test_human_readable_logs.py
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_supported_versions.py
Intermittent failure detected: /marvin/tests/smoke/test_loadbalance.py
Intermittent failure detected: /marvin/tests/smoke/test_network.py
Intermittent failure detected: /marvin/tests/smoke/test_password_server.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dns.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dnsservice.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_iptables_default_policy.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermittent failure detected: /marvin/tests/smoke/test_routers.py
Intermittent failure detected: /marvin/tests/smoke/test_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_usage.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_router_nics.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 57 look OK, 29 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_create_lb_rule_src_nat | `Failure` | 308.00 | test_loadbalance.py
test_02_create_lb_rule_non_nat | `Failure` | 34.66 | test_loadbalance.py
test_assign_and_removal_lb | `Failure` | 35.64 | test_loadbalance.py
test_DeployVmAntiAffinityGroup_in_project | `Error` | 62.32 | test_affinity_groups_projects.py
test_DeployVmAntiAffinityGroup | `Error` | 37.65 | test_affinity_groups.py
test_01_verify_libvirt | `Error` | 602.79 | test_deploy_virtio_scsi_vm.py
test_02_verify_libvirt_after_restart | `Error` | 609.97 | test_deploy_virtio_scsi_vm.py
test_03_verify_libvirt_attach_disk | `Error` | 605.83 | test_deploy_virtio_scsi_vm.py
test_04_verify_guest_lspci | `Error` | 601.99 | test_deploy_virtio_scsi_vm.py
test_05_change_vm_ostype_restart | `Error` | 610.05 | test_deploy_virtio_scsi_vm.py
test_06_verify_guest_lspci_again | `Error` | 602.09 | test_deploy_virtio_scsi_vm.py
ContextSuite context=TestAddConfigtoDeployVM>:setup | `Error` | 0.00 | test_deploy_vm_extra_config_data.py
test_01_port_fwd_on_src_nat | `Failure` | 605.17 | test_network.py
test_02_port_fwd_on_non_src_nat | `Failure` | 607.94 | test_network.py
ContextSuite context=TestPrivateVlansL2Networks>:setup | `Error` | 1230.17 | test_network.py
test_reboot_router | `Failure` | 360.95 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | `Error` | 608.42 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | `Error` | 609.01 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | `Error` | 614.20 | test_network.py
test_01_deployVMInSharedNetwork | `Error` | 659.45 | test_network.py
test_02_verifyRouterIpAfterNetworkRestart | `Error` | 634.51 | test_network.py
test_03_destroySharedNetwork | `Failure` | 1.08 | test_network.py
ContextSuite context=TestSharedNetwork>:teardown | `Error` | 2.14 | test_network.py
test_isolate_network_password_server | `Failure` | 156.99 | test_password_server.py
test_02_vpc_privategw_static_routes | `Failure` | 754.60 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 758.82 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Failure` | 861.23 | test_privategw_acl.py
test_01_disableHumanReadableLogs | `Error` | 602.63 | test_human_readable_logs.py
test_02_enableHumanReadableLogs | `Error` | 602.83 | test_human_readable_logs.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | `Failure` | 279.29 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | `Failure` | 343.17 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 134.20 | test_internal_lb.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 182.11 | test_internal_lb.py
test_01_deploy_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_02_invalid_upgrade_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_03_deploy_and_upgrade_kubernetes_cluster | `Failure` | 0.01 | test_kubernetes_clusters.py
test_04_deploy_and_scale_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_05_delete_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_06_deploy_invalid_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_07_deploy_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_08_deploy_and_upgrade_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_09_delete_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_01_add_delete_kubernetes_supported_version | `Error` | 0.01 | test_kubernetes_supported_versions.py
test_router_dhcphosts | `Failure` | 157.13 | test_router_dhcphosts.py
ContextSuite context=TestRouterDHCPHosts>:teardown | `Error` | 167.41 | test_router_dhcphosts.py
test_router_dhcp_opts | `Error` | 609.23 | test_router_dhcphosts.py
test_router_dns_guestipquery | `Failure` | 454.74 | test_router_dns.py
test_router_dns_guestipquery | `Failure` | 454.71 | test_router_dnsservice.py
test_02_routervm_iptables_policies | `Error` | 661.50 | test_routers_iptables_default_policy.py
test_01_single_VPC_iptables_policies | `Error` | 723.29 | test_routers_iptables_default_policy.py
test_01_isolate_network_FW_PF_default_routes_egress_true | `Failure` | 210.57 | test_routers_network_ops.py
test_02_isolate_network_FW_PF_default_routes_egress_false | `Failure` | 210.61 | test_routers_network_ops.py
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | `Failure` | 250.08 | test_routers_network_ops.py
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | `Failure` | 251.86 | test_routers_network_ops.py
test_03_RVR_Network_check_router_state | `Error` | 696.14 | test_routers_network_ops.py
test_01_router_internal_basic | `Error` | 602.80 | test_routers.py
test_02_router_internal_adv | `Error` | 602.80 | test_routers.py
test_04_restart_network_wo_cleanup | `Error` | 604.84 | test_routers.py
test_01_service_offering_cpu_limit_use | `Error` | 100.51 | test_service_offerings.py
test_04_change_offering_small | `Failure` | 708.08 | test_service_offerings.py
test_01_snapshot_root_disk | `Error` | 606.03 | test_snapshots.py
test_03_ssvm_internals | `Error` | 602.81 | test_ssvm.py
test_04_cpvm_internals | `Error` | 602.92 | test_ssvm.py
test_05_stop_ssvm | `Error` | 680.62 | test_ssvm.py
test_06_stop_cpvm | `Error` | 653.67 | test_ssvm.py
test_07_reboot_ssvm | `Error` | 718.17 | test_ssvm.py
test_08_reboot_cpvm | `Error` | 627.95 | test_ssvm.py
test_09_reboot_ssvm_forced | `Error` | 739.74 | test_ssvm.py
test_10_reboot_cpvm_forced | `Error` | 651.90 | test_ssvm.py
test_11_destroy_ssvm | `Error` | 673.81 | test_ssvm.py
test_12_destroy_cpvm | `Error` | 669.40 | test_ssvm.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
test_10_attachAndDetach_iso | `Failure` | 607.70 | test_vm_life_cycle.py
test_01_create_vm_snapshots | `Failure` | 604.48 | test_vm_snapshots.py
test_02_revert_vm_snapshots | `Failure` | 601.43 | test_vm_snapshots.py
test_03_delete_vm_snapshots | `Failure` | 0.02 | test_vm_snapshots.py
test_01_create_volume | `Failure` | 607.74 | test_volumes.py
test_02_attach_volume | `Failure` | 606.62 | test_volumes.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 833.98 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 831.92 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 749.19 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Error` | 716.04 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 809.34 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 836.25 | test_vpc_redundant.py
test_01_VPC_nics_after_destroy | `Failure` | 722.20 | test_vpc_router_nics.py
test_02_VPC_default_routes | `Failure` | 724.40 | test_vpc_router_nics.py
test_01_redundant_vpc_site2site_vpn | `Failure` | 376.62 | test_vpc_vpn.py
test_01_vpc_site2site_vpn_multiple_options | `Error` | 251.18 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Error` | 282.78 | test_vpc_vpn.py
test_hostha_enable_ha_when_host_disconected | `Error` | 605.03 | test_hostha_kvm.py
test_hostha_kvm_host_degraded | `Failure` | 669.38 | test_hostha_kvm.py
test_hostha_kvm_host_fencing | `Failure` | 644.98 | test_hostha_kvm.py
test_hostha_kvm_host_recovering | `Failure` | 643.97 | test_hostha_kvm.py

blueorangutan commented:
<b>[S] Trillian test result (tid-189)</b>
Environment: kvm-ubuntu18 (x2), Advanced Networking with Mgmt server u18
Total time taken: 112373 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr4567-t189-kvm-ubuntu18.zip
Intermittent failure detected: /marvin/tests/smoke/test_deploy_virtio_scsi_vm.py
Intermittent failure detected: /marvin/tests/smoke/test_deploy_vm_extra_config_data.py
Intermittent failure detected: /marvin/tests/smoke/test_human_readable_logs.py
Intermittent failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_clusters.py
Intermittent failure detected: /marvin/tests/smoke/test_kubernetes_supported_versions.py
Intermittent failure detected: /marvin/tests/smoke/test_loadbalance.py
Intermittent failure detected: /marvin/tests/smoke/test_network.py
Intermittent failure detected: /marvin/tests/smoke/test_password_server.py
Intermittent failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dhcphosts.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dns.py
Intermittent failure detected: /marvin/tests/smoke/test_router_dnsservice.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_iptables_default_policy.py
Intermittent failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermittent failure detected: /marvin/tests/smoke/test_routers.py
Intermittent failure detected: /marvin/tests/smoke/test_service_offerings.py
Intermittent failure detected: /marvin/tests/smoke/test_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_ssvm.py
Intermittent failure detected: /marvin/tests/smoke/test_usage.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermittent failure detected: /marvin/tests/smoke/test_vm_snapshots.py
Intermittent failure detected: /marvin/tests/smoke/test_volumes.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_router_nics.py
Intermittent failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Intermittent failure detected: /marvin/tests/smoke/test_host_maintenance.py
Intermittent failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 60 look OK, 26 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_create_lb_rule_src_nat | `Failure` | 309.68 | test_loadbalance.py
test_02_create_lb_rule_non_nat | `Failure` | 35.76 | test_loadbalance.py
test_assign_and_removal_lb | `Failure` | 35.65 | test_loadbalance.py
test_01_verify_libvirt | `Error` | 603.97 | test_deploy_virtio_scsi_vm.py
test_02_verify_libvirt_after_restart | `Error` | 611.00 | test_deploy_virtio_scsi_vm.py
test_03_verify_libvirt_attach_disk | `Error` | 606.14 | test_deploy_virtio_scsi_vm.py
test_04_verify_guest_lspci | `Error` | 602.06 | test_deploy_virtio_scsi_vm.py
test_05_change_vm_ostype_restart | `Error` | 614.39 | test_deploy_virtio_scsi_vm.py
test_06_verify_guest_lspci_again | `Error` | 602.11 | test_deploy_virtio_scsi_vm.py
ContextSuite context=TestAddConfigtoDeployVM>:setup | `Error` | 0.00 | test_deploy_vm_extra_config_data.py
test_01_port_fwd_on_src_nat | `Failure` | 605.54 | test_network.py
test_02_port_fwd_on_non_src_nat | `Failure` | 608.15 | test_network.py
ContextSuite context=TestPrivateVlansL2Networks>:setup | `Error` | 1231.58 | test_network.py
test_reboot_router | `Failure` | 365.99 | test_network.py
test_network_rules_acquired_public_ip_1_static_nat_rule | `Error` | 610.03 | test_network.py
test_network_rules_acquired_public_ip_2_nat_rule | `Error` | 610.63 | test_network.py
test_network_rules_acquired_public_ip_3_Load_Balancer_Rule | `Error` | 615.10 | test_network.py
test_01_deployVMInSharedNetwork | `Error` | 675.26 | test_network.py
test_02_verifyRouterIpAfterNetworkRestart | `Error` | 634.53 | test_network.py
test_03_destroySharedNetwork | `Failure` | 1.09 | test_network.py
ContextSuite context=TestSharedNetwork>:teardown | `Error` | 2.21 | test_network.py
test_isolate_network_password_server | `Failure` | 157.20 | test_password_server.py
test_02_vpc_privategw_static_routes | `Failure` | 765.29 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 763.28 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Failure` | 854.37 | test_privategw_acl.py
test_01_disableHumanReadableLogs | `Error` | 604.85 | test_human_readable_logs.py
test_02_enableHumanReadableLogs | `Error` | 604.81 | test_human_readable_logs.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | `Failure` | 274.49 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | `Failure` | 324.70 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 145.50 | test_internal_lb.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | `Error` | 185.34 | test_internal_lb.py
test_01_deploy_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_02_invalid_upgrade_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_03_deploy_and_upgrade_kubernetes_cluster | `Failure` | 0.01 | test_kubernetes_clusters.py
test_04_deploy_and_scale_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_05_delete_kubernetes_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_06_deploy_invalid_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_07_deploy_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_08_deploy_and_upgrade_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_09_delete_kubernetes_ha_cluster | `Failure` | 0.00 | test_kubernetes_clusters.py
test_01_add_delete_kubernetes_supported_version | `Error` | 0.01 | test_kubernetes_supported_versions.py
test_router_dhcphosts | `Failure` | 157.40 | test_router_dhcphosts.py
ContextSuite context=TestRouterDHCPHosts>:teardown | `Error` | 168.79 | test_router_dhcphosts.py
test_router_dhcp_opts | `Error` | 610.47 | test_router_dhcphosts.py
test_router_dns_guestipquery | `Failure` | 455.08 | test_router_dns.py
test_router_dns_guestipquery | `Failure` | 455.45 | test_router_dnsservice.py
test_02_routervm_iptables_policies | `Error` | 663.90 | test_routers_iptables_default_policy.py
test_01_single_VPC_iptables_policies | `Error` | 727.92 | test_routers_iptables_default_policy.py
test_01_isolate_network_FW_PF_default_routes_egress_true | `Failure` | 212.10 | test_routers_network_ops.py
test_02_isolate_network_FW_PF_default_routes_egress_false | `Failure` | 212.47 | test_routers_network_ops.py
test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true | `Failure` | 249.00 | test_routers_network_ops.py
test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false | `Failure` | 253.75 | test_routers_network_ops.py
test_03_RVR_Network_check_router_state | `Error` | 696.97 | test_routers_network_ops.py
test_01_router_internal_basic | `Error` | 603.62 | test_routers.py
test_02_router_internal_adv | `Error` | 603.61 | test_routers.py
test_04_restart_network_wo_cleanup | `Error` | 606.09 | test_routers.py
test_01_service_offering_cpu_limit_use | `Error` | 100.79 | test_service_offerings.py
test_04_change_offering_small | `Failure` | 716.39 | test_service_offerings.py
test_01_snapshot_root_disk | `Error` | 607.93 | test_snapshots.py
test_03_ssvm_internals | `Error` | 604.39 | test_ssvm.py
test_04_cpvm_internals | `Error` | 604.21 | test_ssvm.py
test_05_stop_ssvm | `Error` | 693.59 | test_ssvm.py
test_06_stop_cpvm | `Error` | 665.50 | test_ssvm.py
test_07_reboot_ssvm | `Error` | 722.43 | test_ssvm.py
test_08_reboot_cpvm | `Error` | 629.89 | test_ssvm.py
test_09_reboot_ssvm_forced | `Error` | 760.96 | test_ssvm.py
test_10_reboot_cpvm_forced | `Error` | 650.94 | test_ssvm.py
test_11_destroy_ssvm | `Error` | 659.09 | test_ssvm.py
test_12_destroy_cpvm | `Error` | 659.86 | test_ssvm.py
test_01_migrate_VM_and_root_volume | `Error` | 84.52 | test_vm_life_cycle.py
test_02_migrate_VM_with_two_data_disks | `Error` | 68.36 | test_vm_life_cycle.py
test_01_secure_vm_migration | `Error` | 731.15 | test_vm_life_cycle.py
test_02_unsecure_vm_migration | `Error` | 643.56 | test_vm_life_cycle.py
test_03_secured_to_nonsecured_vm_migration | `Error` | 639.77 | test_vm_life_cycle.py
test_04_nonsecured_to_secured_vm_migration | `Error` | 639.04 | test_vm_life_cycle.py
test_08_migrate_vm | `Error` | 46.14 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | `Failure` | 607.66 | test_vm_life_cycle.py
test_01_create_vm_snapshots | `Failure` | 604.66 | test_vm_snapshots.py
test_02_revert_vm_snapshots | `Failure` | 601.61 | test_vm_snapshots.py
test_03_delete_vm_snapshots | `Failure` | 0.02 | test_vm_snapshots.py
test_01_create_volume | `Failure` | 610.23 | test_volumes.py
test_02_attach_volume | `Failure` | 605.79 | test_volumes.py
test_01_create_redundant_VPC_2tiers_4VMs_4IPs_4PF_ACL | `Error` | 837.34 | test_vpc_redundant.py
test_02_redundant_VPC_default_routes | `Error` | 839.97 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Error` | 764.30 | test_vpc_redundant.py
test_04_rvpc_network_garbage_collector_nics | `Error` | 713.97 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 0.50 | test_vpc_redundant.py
test_01_VPC_nics_after_destroy | `Failure` | 732.24 | test_vpc_router_nics.py
test_02_VPC_default_routes | `Failure` | 728.21 | test_vpc_router_nics.py
test_01_redundant_vpc_site2site_vpn | `Failure` | 369.94 | test_vpc_vpn.py
test_01_vpc_site2site_vpn_multiple_options | `Error` | 256.98 | test_vpc_vpn.py
test_01_vpc_site2site_vpn | `Error` | 297.65 | test_vpc_vpn.py
test_hostha_enable_ha_when_host_disconected | `Error` | 605.87 | test_hostha_kvm.py
test_hostha_kvm_host_degraded | `Error` | 674.35 | test_hostha_kvm.py
test_hostha_kvm_host_fencing | `Error` | 646.22 | test_hostha_kvm.py
test_hostha_kvm_host_recovering | `Error` | 646.88 | test_hostha_kvm.py

DaanHoogland commented:
> Tests indicate that all `LibvirtException` have been handled. LGTM.

@GabrielBrascher I merged based on this comment of yours, but then we had some internal discussion on whether it was tested enough. I don't think any of the failures above are due to this change, but still, can you comment a bit more on testing please? cc @GutoVeronezi
